src/spac/visualization.py

```diff
@@ -77,29 +85,62 @@ def visualize_2D_scatter(
     }
 
     if theme and theme not in themes:
-        error_msg = (
-            f"Theme '{theme}' not recognized. Please use a valid theme."
-        )
+        error_msg = f"Theme '{theme}' not recognized. Please use a valid theme."
         raise ValueError(error_msg)
+
     cmap = themes.get(theme, plt.get_cmap('viridis'))
 
     # Determine point size
     num_points = len(x)
     if point_size is None:
         point_size = 5000 / num_points
 
+    # Get figure size from kwargs or set defaults
+    fig_width = kwargs.get('fig_width', 10)
+    fig_height = kwargs.get('fig_height', 8)
+    fontsize = kwargs.get('fontsize', 12)
+
     if ax is None:
-        fig, ax = plt.subplots(figsize=(10, 6))
+        fig, ax = plt.subplots(figsize=(fig_width, fig_height))
     else:
         fig = ax.figure
 
-    # Plotting logic
-    if labels is not None:
+    labels = kwargs.pop('labels', None)
+    if labels is not None and len(labels) != len(x):
+        raise ValueError("The length of labels must match the length of x and y.")

```

Comment:
this is the source of why one of the unittests are failing. 

It's expecting an error message of  "Labels length should match x and y length." for the unittest test_labels_length_mismatch

You can change this error message but you'd also have to up date the unittest to match this error message.